File: jsimple-logging-stdimpl/c#/jsimple/logging/stdimpl/WriterAppender.cs
using System;

namespace jsimple.logging.stdimpl
{

	using Writer = jsimple.io.Writer;
	using PlatformUtils = jsimple.util.PlatformUtils;



	/// <summary>
	/// @author Bret Johnson
	/// @since 4/11/13 4:10 PM
	/// </summary>
	public class WriterAppender : Appender
	{
		internal Writer writer;
		internal bool flushImmediately = true;

		public WriterAppender(Writer writer)
		{
			this.writer = writer;
		}

		public override void append(LoggingEvent loggingEvent)
		{
			lock (this)
			{
				writer.writeln(loggingEvent.Level.DefaultDisplayName + " " + loggingEvent.LoggerName + " - " + loggingEvent.FormattedMessage);

				Exception throwable = loggingEvent.Throwable;
				if (throwable != null)
					writer.writeln(PlatformUtils.getExceptionDescription(throwable));

				if (flushImmediately)
					writer.flush();
			}

			/*
			#logback.classic pattern: %d [%thread] %-5level %logger{36} - %msg%n
			2012-04-26 14:54:38,461 [main] DEBUG com.foo.App - Hello world
			2012-04-26 14:54:38,461 [main] DEBUG com.foo.App - Hi again
			*/

			//To change body of implemented methods use File | Settings | File Templates.
		}
	}

}

File: jsimple-logging-stdimpl/src/main/java/jsimple/logging/stdimpl/WriterAppender.java
/*
 * Copyright (c) 2012-2014 Microsoft Mobile.  All Rights Reserved.
 *
 * This file is based on or incorporates material from Logback http://logback.qos.ch
 * (collectively, “Third Party Code”). Microsoft Mobile is not the original
 * author of the Third Party Code.
 *
 * This program and the accompanying materials are made available under the terms
 * of the Eclipse Public License v1.0 which accompanies this distribution, and is
 * available at http://www.eclipse.org/legal/epl-v10.html.
 */

package jsimple.logging.stdimpl;

import jsimple.io.Writer;
import jsimple.util.PlatformUtils;
import org.jetbrains.annotations.Nullable;

/**
 * @author Bret Johnson
 * @since 4/11/13 4:10 PM
 */
public class WriterAppender extends Appender {
    Writer writer;
    boolean flushImmediately = true;

    public WriterAppender(Writer writer) {
        this.writer = writer;
    }

    @Override public void append(LoggingEvent loggingEvent) {
        synchronized (this) {
            writer.writeln(loggingEvent.getLevel().getDefaultDisplayName() + " " + loggingEvent.getLoggerName() + " - " +
                    loggingEvent.getFormattedMessage());

            @Nullable Throwable throwable = loggingEvent.getThrowable();
            if (throwable != null)
                writer.writeln(PlatformUtils.getExceptionDescription(throwable));

            if (flushImmediately)
                writer.flush();
        }

        /*
        #logback.classic pattern: %d [%thread] %-5level %logger{36} - %msg%n
        2012-04-26 14:54:38,461 [main] DEBUG com.foo.App - Hello world
        2012-04-26 14:54:38,461 [main] DEBUG com.foo.App - Hi again
        */

        //To change body of implemented methods use File | Settings | File Templates.
    }
}
